rosbag_snapshot/src/snapshotter.cpp

```diff
@@ -87,6 +94,71 @@ bool SnapshotterOptions::addTopic(std::string const& topic, ros::Duration durati
   return ret.second;
 }
 
+bool SnapshotterOptions::addPattern(std::string const& pattern, ros::Duration duration, int32_t memory, int32_t count)
+{
+  SnapshotterTopicOptions ops(duration, memory, count);
+  try
+  {
+    patterns_.emplace_back(std::make_shared<SnapshotterTopicPattern>(pattern, ops));
+    ROS_INFO("Added topic pattern: %s", pattern.c_str());
+    return true;
+  }
+  catch (boost::regex_error& e)
+  {
+    ROS_ERROR("Invalid topic pattern '%s': %s", pattern.c_str(), e.what());
+    return false;
+  }
+}
+
+bool SnapshotterOptions::addTopicOrPattern(std::string const& topic_or_pattern,
+                                           ros::Duration duration, int32_t memory, int32_t count)
+{
+  if (is_topic_name_pattern(topic_or_pattern))
+    return addPattern(topic_or_pattern, duration, memory, count);
+  else
+    return addTopic(topic_or_pattern, duration, memory, count);
+}
+
+SnapshotterTopicPatternConstPtr SnapshotterOptions::findFirstMatchingPattern(const std::string &topic)
+{
+  if (patterns_.empty())
+    return nullptr;
+  auto it = matching_patterns_cache_.find(topic);
+  if (it != matching_patterns_cache_.end())
+    return it->second;
+  SnapshotterTopicPatternConstPtr matching_pattern = nullptr;
+  for (auto& pattern : patterns_)
```

Comment:
Can you please put in curly braces for the for loop? Its not a strict requirement, I just dislike the look when its multiple lines (even though its one statement)